libs/tracing/src/main/kotlin/net/corda/tracing/TraceThis.kt

```diff
@@ -44,18 +46,72 @@ fun getOrCreateBatchPublishTracing(clientId: String): BatchPublishTracing {
     return TracingState.currentTraceService.getOrCreateBatchPublishTracing(clientId)
 }
 
-fun addTraceContextToRecords(records: List<Record<*, *>>): List<Record<*, *>> = records.map(::addTraceContextToRecord)
+fun addTraceContextToRecords(
+    records: List<Record<*, *>>
+): List<Record<*, *>> = records.map(::addTraceContextToRecord)
+
+fun addTraceContextToRecord(
+    record: Record<*, *>
+): Record<out Any, out Any> {
+    return record.copy(
+        headers = TracingState.currentTraceService.addTraceHeaders(
+            record.headers,
+            emptyList() // Don't override the current trace context
+        )
+    )
+}
+
+fun addTraceContextToRecord(
+    record: Record<*, *>,
+    traceHeadersToOverrideContext: Map<String, Any> = emptyMap() // By default, don't override the current trace context
+): Record<out Any, out Any> {
```

Comment:
If we make this change we can remove the cast at the call site:

```suggestion
fun <K : Any, E : Any> addTraceContextToRecord(
    record: Record<K, E>,
    traceHeadersToOverrideContext: Map<String, Any> = emptyMap() // By default, don't override the current trace context
): Record<K, E> {
```